Exiled.Events/Patches/Events/Player/PreAuthenticating.cs

```diff
@@ -33,105 +37,113 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
-                    new(OpCodes.Ldstr, "Player {0} tried to preauthenticated from endpoint {1}, but the request has been rejected by a plugin."),
-                    new(OpCodes.Ldloc_S, 10),
-                    new(OpCodes.Ldarg_1),
-                    new(OpCodes.Ldfld, Field(typeof(ConnectionRequest), nameof(ConnectionRequest.RemoteEndPoint))),
-                    new(OpCodes.Call, Method(typeof(string), nameof(string.Format), new[] { typeof(string), typeof(object), typeof(object) })),
-                    new(OpCodes.Dup),
-                    new(OpCodes.Stloc_S, failedMessage.LocalIndex),
+                    // If server full
+                    new(OpCodes.Ldloc, jumpConditions.LocalIndex),
+                    new(OpCodes.Ldc_I4_1),
+                    new(OpCodes.Beq, fullRejectLabel),
 
-                    // ServerConsole.AddLog(failedMessage, ConsoleColor.Gray)
-                    new(OpCodes.Ldc_I4_7),
-                    new(OpCodes.Call, Method(typeof(ServerConsole), nameof(ServerConsole.AddLog))),
+                    // If fully reject
+                    new(OpCodes.Ldloc, jumpConditions.LocalIndex),
+                    new(OpCodes.Ldc_I4_M1),
+                    new(OpCodes.Beq, returnLabel),
 
-                    // ServerLogs.AddLog(ServerLogs.Modules.Networking, failedMessage, ServerLogs.ServerLogType.ConnectionUpdate, false)
-                    new(OpCodes.Ldc_I4_1),
-                    new(OpCodes.Ldloc_S, failedMessage.LocalIndex),
-                    new(OpCodes.Ldc_I4_0),
-                    new(OpCodes.Ldc_I4_0),
-                    new(OpCodes.Call, Method(typeof(ServerLogs), nameof(ServerLogs.AddLog))),
-                    new(OpCodes.Br_S, returnLabel),
                 });
 
             for (int z = 0; z < newInstructions.Count; z++)
                 yield return newInstructions[z];
 
             ListPool<CodeInstruction>.Shared.Return(newInstructions);
         }
+
+        public static int authenticatePlayer(string userId, ConnectionRequest request, byte flags, string country, int cur_slots)
+        {
+            PreAuthenticatingEventArgs ev = new(userId, request, request.Data.Position,  flags, country, cur_slots);
+
+            Player.OnPreAuthenticating(ev);
+
+            // If there is a desire to accept the connection naturally (server not full, and accept condition left alone)
+            if (ev.AcceptConnection)
+            {
+                return 2;
+            }
+
+            // If the server wants the rest of NW condition checks
+            if (ev.AllowFurtherChecks)
+            {
+                return 0;
+            }
+
+            // If the connection should be accepted due to ForceAcceptConnection default/
+            if (ev.ForceAllowConnection)
+            {
+                return 2;
+            }
```

Comment:
The entire ForceAllowConnection won't allow the player to join if the event has been cancelled by an EXILED plugin. The player will be kicked anyway and we will end with a malformed preauth response being sent by the server and redundant transport layer operations.

Once the event is cancelled by the plugin, it can't be uncancelled. It was even recently tested.